### PR TITLE
*: Fix some missing error checks as reported by golangci-lint

### PR DESCRIPTION
**Pull request by suzaku**
### What problem does this PR solve? 

Fix some missing error checks as reported by golangci-lint

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by july2993** [comment]
pls fix ci, and this ps fix `some` not `all`?

**Comment by suzaku** [comment]
> pls fix ci, and this ps fix `some` not `all`?

Yes, this PR only want to fix some of the easy ones.

The others are more complex, some involves changing the workflow and others are just result of bad design of the library API.
For the first kind, we can fix them in separate PRs instead of hiding them among other simple fixes.
For the second kind, I don't think we want to fix them at all.

**Comment by WangXiangUSTC** [comment]
LGTM

File: pump/storage/log.go [hunk 1]
```diff
@@ -338,7 +338,13 @@ func seekToNextRecord(reader *bufio.Reader) (bytes int, err error) {
 			return
 		}
 
-		reader.Discard(1)
+		var discarded int
+		discarded, err = reader.Discard(1)
+		if err != nil {
```
Comment (july2993) [inline pump/storage/log.go]: At L329
we `reader.Peek(4)` and success, so we should always success to Discard one here.
what about just fatal if `err != nil` here and add some comment

Comment (suzaku) [reply pump/storage/log.go]: I'm not sure. What's the difference it should never happens?

**Pull request merged**